evm/src/contracts/CircleIntegration/Governance.sol

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.22;
+
+import {ERC1967Upgrade} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+import {BytesParsing} from "src/libraries/BytesParsing.sol";
+import {IWormhole} from "src/interfaces/IWormhole.sol";
+import {IGovernance} from "src/interfaces/IGovernance.sol";
+
+import {State} from "./State.sol";
+import {
+    SequentialRegistration,
+    getRegisteredEmitters,
+    getChainToDomain,
+    getConsumedVaas,
+    getDomainToChain,
+    getSequentialRegistration
+} from "./Storage.sol";
+
+abstract contract Governance is IGovernance, State, ERC1967Upgrade {
+    using BytesParsing for bytes;
+
+    error UnsupportedGovernanceAction();
+
+    uint16 constant GOVERNANCE_CHAIN = 1;
+    bytes32 constant GOVERNANCE_EMITTER =
+        0x0000000000000000000000000000000000000000000000000000000000000004;
+
+    // "CircleIntegration" (left-zero-padded)
+    bytes32 constant GOVERNANCE_MODULE =
+        0x000000000000000000000000000000436972636c65496e746567726174696f6e;
+
+    uint8 constant UPDATE_WORMHOLE_FINALITY = 1;
+    uint8 constant REGISTER_EMITTER_AND_DOMAIN = 2;
+    uint8 constant CONTRACT_UPGRADE = 3;
+
+    /**
+     * @notice `updateWormholeFinality` changes the wormhole message consistencyLevel.
+     * @param encodedVaa Attested Wormhole governance message with the following
+     * relevant fields:
+     * - Field Bytes Type Index
+     * - finality 1 uint8 35
+     */
+    function updateWormholeFinality(bytes memory encodedVaa) public pure {
```

Comment:
Is there any reason to keep this method around?

Reply:
Removed here: https://github.com/wormhole-foundation/wormhole-circle-integration/pull/48/commits/1cfd94c8316434d83e5816779bc968c972a65bb5